test/common/network/address_impl_test.cc

```diff
@@ -466,15 +466,15 @@ TEST(AddressFromSockAddrDeathTest, IPv4) {
   EXPECT_EQ(1, inet_pton(AF_INET, "1.2.3.4", &sin.sin_addr));
   sin.sin_port = htons(6502);
 
-  EXPECT_DEATH(addressFromSockAddr(ss, 1), "ss_len");
-  EXPECT_DEATH(addressFromSockAddr(ss, sizeof(sockaddr_in) - 1), "ss_len");
-  EXPECT_DEATH(addressFromSockAddr(ss, sizeof(sockaddr_in) + 1), "ss_len");
+  EXPECT_DEATH(*addressFromSockAddr(ss, 1), "ss_len");
```

Comment:
Rather than EXPECT_DEATH can you just do
```
   absl::Status status = addressFromSockAddr(ss, 1);
   EXPECT_FALSE(status.ok());
   EXPECT_THAT(status.message()).Contains("ss_len")
```
I might not have that syntax right. But I think that would be a clearer test than a death-test, and would also be faster.

You could also make a macro EXPECT_FAILURE_CONTAINS(status, message_substring) so that it was equally terse from current state.

Reply:
These will trigger RELEASE_ASSERT inside addressFromSockAddr, won't return any status code.

Reply:
Then should that be renamed addressFromSockAddrOrDie?

Reply:
wait...then why did you need to add the `*`? It looks to me like addressFromSockAddr must be returning then, otherwise the `*` wouldn't run and you would not have bothered to add it.

Reply:
StatusOr return value is ABSL_MUST_USE_RESULT and must be used as required by the compiler.

Reply:
it has release_assert, but it may return errors at the same time.

Reply:
That is very confusing behavior. WDYT of renaming it addressFromSockAddrOrDie and changing its return-type back to InstanceConstSharedPtr?

Was there a case previously where it could return null?

Reply:
Oh I see there *was* a case where it would throw, which you turned into an error. Is that case hit with a test? Maybe we could turn that into another RELEASE_ASSERT.

Reply:
There are several places where error status may be returned, createInstancePtr or unexpected address family in the end of switch  block.

Reply:
I pushed a new commit, now we have addressFromSockAddr and addressFromSockAddrOrThrow